src/Skeletor/Packages/Interfaces/PreInstallPackageInterface.php

```diff
@@ -0,0 +1,7 @@
+<?php
+namespace Skeletor\Packages\Interfaces;
+
+interface PreInstallPackageInterface
```

Comment:
Should it extend the PackageInterface? Do you foresee a scenario where you run preInstall and not Install? Seems reasonable to assume that if you use PreInstall you will also need the Install interface, this would save from implementing two interfaces in a concrete class

Reply:
@titopixelfusion In this case all the packages extend the `Package` class which implements the `PackageInterface` so a package itself never has to implement the interface and only to more specific interfaces like the `PreInstallPackageInterface`. Do you think we should refactor this?

Reply:
It is not a big concern, I am happy to leave things as they are, we can always change later the  PreInstallPackageInterface to extend InstallPackageInterface, it doesn't impact the current design.